### PR TITLE
Doc: Add tip for checking for existing field

### PR DESCRIPTION
**Pull request by karenzone**
Co-authored-by: Sebastian <gentunian@users.noreply.github.com>

## Release notes
Doc: Add tip for how to see if a boolean field exists

## What does this PR do?
Provides a work-around for checking to see if a boolean field exists (long standing issue)

## Related issues
- Relates #1867

## Preview
https://logstash_12899.docs-preview.app.elstc.co/guide/en/logstash/master/tips.html#tip-filters

### CODE REVIEW

**Comment by karenzone** [comment]
@gentunian, thank you for sharing this info with the community. I welcome your input on both the content and the attribution. Please let me know if you would like to see any changes.

**Comment by gentunian** [comment]
@karenzone looks good to me. By the way, almost 1.5 years passed since I managed to use that workaround. Did you checked that this still works in current logstash versions? I think it should, as it makes use of simple logic and `@metadata` fields. But I went away of logstash a long time ago since I switched projects.

**Comment by karenzone** [comment]
Thanks, @gentunian.  We will validate as part of the review process.  
We appreciate your contribution and hope that you'll be back using Logstash soon!  ❤️

File: docs/static/best-practice.asciidoc [hunk 1]
```diff
@@ -55,15 +55,57 @@ no longer specify local pipeline configurations. The `pipelines.yml` file and
...
+  # now we now if testField didn't exists, our field will have 
+  # the initial arbitrary value
+  if [@metadata][testField_check] == "unknown arbitrary value" {
+
+    # just for debugging purpouses...
+    mutate { add_field => { "FIELD_DID_NOT_EXISTED" => true }}
+
+  } else {
+    # just for debugging purpouses...
+    mutate { add_field => { "FIELD_DID_ALREADY_EXISTED" => true }}
```
Comment (kares) [inline docs/static/best-practice.asciidoc:95]:
```suggestion
    # logic to execute when [test_field] existed
    mutate { add_field => { "field_did_exist" => true }}
```

File: docs/static/best-practice.asciidoc [hunk 2]
```diff
@@ -55,15 +55,57 @@ no longer specify local pipeline configurations. The `pipelines.yml` file and
...
+  }
+
+
+  # now we now if testField didn't exists, our field will have 
+  # the initial arbitrary value
+  if [@metadata][testField_check] == "unknown arbitrary value" {
+
+    # just for debugging purpouses...
+    mutate { add_field => { "FIELD_DID_NOT_EXISTED" => true }}
+
```
Comment (kares) [inline docs/static/best-practice.asciidoc:92]:
```suggestion
    # logic to execute when [test_field] did not exist
    mutate { add_field => { "field_did_not_exist" => true }}
```

File: docs/static/best-practice.asciidoc [hunk 3]
```diff
@@ -55,15 +55,57 @@ no longer specify local pipeline configurations. The `pipelines.yml` file and
...
+-----
+filter {
+  mutate {
+    # we use a "temporal" field with a predefined arbitrary known value that
+    # lives only in filtering stage.
+    add_field => { "[@metadata][testField_check]" => "unknown arbitrary value" }
+
+    # we copy the field of interest into that temporal field.
+    # If the field doesn't exist, copy is not executed.
+    copy => { "testField" => "[@metadata][testField_check]" }
```
Comment (kares) [inline docs/static/best-practice.asciidoc:82]:
```suggestion
    copy => { "test_field" => "[@metadata][test_field_check]" }
```

File: docs/static/best-practice.asciidoc [hunk 4]
```diff
@@ -55,15 +55,57 @@ no longer specify local pipeline configurations. The `pipelines.yml` file and
...
+  }
+
+
+  # now we now if testField didn't exists, our field will have 
+  # the initial arbitrary value
+  if [@metadata][testField_check] == "unknown arbitrary value" {
```
Comment (kares) [inline docs/static/best-practice.asciidoc:88]:
```suggestion
  if [@metadata][test_field_check] == "a null value" {
```

**Review by kares — COMMENTED** [review]
looking good, would rather use a consistent field naming convention as suggested
(dashed_lower_case aligns with ECS naming conventions)

File: docs/static/best-practice.asciidoc [hunk 5]
```diff
@@ -55,15 +55,55 @@ no longer specify local pipeline configurations. The `pipelines.yml` file and
...
+{ls} supports [@metadata] fields--fields that are not visible for output plugins and live only in the filtering state. 
+You can use [@metadata] fields with the mutate filter to see if a field exists. 
+
+[source,ruby]
+-----
+filter {
+  mutate {
+    # we use a "temporal" field with a predefined arbitrary known value that
+    # lives only in filtering stage.
+    add_field => { "[@metadata][testField_check]" => "unknown arbitrary value" }
```
Comment (kares) [inline docs/static/best-practice.asciidoc:78]:
```suggestion
    add_field => { "[@metadata][test_field_check]" => "a null value" }
```

**Review by kares — APPROVED** [review]
:icecream:

**Review by karenzone — COMMENTED** [review]
Builds cleanly and renders as expected

**Pull request merged**